src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -224,6 +224,11 @@ public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax no
                 var oldMethod = _containingMemberOrLambda;
                 _containingMemberOrLambda = match;
 
+                foreach (var parameter in node.ParameterList.Parameters)
+                {
+                    Visit(parameter.Default);
```

Comment:
Default expressions are never bound directly with a binder from the factory or the cache. They are _always_ wrapped by a binder in `MakeDefaultExpression` before the expression is bound. So it appears to be a consistency rather than an inconsistency.